pkg/identity/issuer.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identity
+
+import "context"
+
+type Issuer interface {
+	// Can this Issuer authenticate the given issuer URL?
+	Match(ctx context.Context, url string) bool
+
+	// Authenticate ID token and return Principal on success
+	Authenticate(ctx context.Context, token string) (Principal, error)
```

Comment:
To clarify, does this just extract the principal, or also do signature verification of the token? If both, we should mention that in the comment. Though I think verification of the token should happen early on in the request flow.

Reply:
extraction and signature verification. Basically the call to `IssuerPool.Authenticate()` will replace this line right here https://github.com/sigstore/fulcio/blob/a08e6a73669b12255fd5dab6f7851f327dfdc56e/pkg/api/grpc_server.go#L73

Reply:
The comment does say the token is authenticated (so the signature is checked) and a `Principal` is returned. I can clarify that authenticating an ID token means checking the signature if that makes it more clear?

Reply:
SG on replacing that line. That'd be great to clarify that this verifies the signature too.